src/metabase/upload.clj

```diff
@@ -347,13 +357,14 @@
 (defn- parse-rows
   "Returns a lazy seq of parsed rows, given a sequence of upload types for each column.
   Empty strings are parsed as nil."
-  [col-upload-types rows]
-  (let [settings (upload-parsing/get-settings)
-        parsers  (map #(upload-parsing/upload-type->parser % settings) col-upload-types)]
-    (for [row rows]
-      (for [[value parser] (u/map-all vector row parsers)]
-        (when-not (str/blank? value)
-          (parser value))))))
+  ([col-upload-types rows]
+   (parse-rows (upload-parsing/get-settings) col-upload-types rows))
```

Comment:
Why the change?

Reply:
1. By passing this in the 3-arity function becomes pure. I like purity 😄 
2. There's a bit of work to get the settings, so it's wasteful to fetch it multiple times.

An earlier version of this was passing it in somewhere, but I see that's gone. There's still one spot I meant to update and missed.